AEPMessaging/Sources/Messaging+EventHistory.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ Copyright 2021 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPCore
+import AEPServices
+import Foundation
+
+extension Messaging {
```

Comment:
I see this as standalone functionality, and it can be it's own helper class, not necessarily an extension to Messaging and instead of passing over the runtime, you could use the MobileCore.dispatch API.
You could have a class e.g. MessagingEventHistory and have something like `MessagingEventHistory.record(params)`, or `MessagingEventHistory.recordInApp(params)` if you anticipate different types of messages to be recorded here in the future.

Reply:
i could - why is that strategy better?